docs/src/docs/asciidoc/user/clustered-cache.adoc

```diff
@@ -36,50 +62,92 @@ Check for the below `INFO` log to confirm if the server started successfully,
-    creates the clustered storage in the server if it doesn't already exist.
+    Sample URI provided in the example is pointing to the clustered storage named `my-application` on the Terracotta server (Assuming the server is running on localhost and port *9510*).
+    The query-param `auto-create` would make sure that the cache manager creates the clustered storage instance named `my-application` in the server if it doesn't already exist.
 <3> Returns a fully initialized cache manager that can be used to create clustered caches.
 <4> Close the cache manager.
 
 == Cache manager configuration and usage of server side resources
 
-After starting the Terracotta server (Refer section: <<Starting Terracotta server>>), Cache manager with clustering capabilities can be configured with server side resource(s) and in turn can be used by cache(s) :
+After starting the Terracotta server (Refer section: <<start-server, Starting Terracotta server>>), cache managers with clustering capabilities can be configured with server side resource(s) which in turn can be used by cache(s).
+The following sample; unlike the previous one; uses a broader clustering service configuration that lets you configure more things at the cache manager level.
+Before we jump to configuration samples, we'll define certain terms that are used in the remainder of this document.
+
+Server off-heap resource::
+Server off-heap resources are storage resources defined at the server.
+`primary-server-resource` and `secondary-server-resource` are two named server off-heap resources defined in the server configuration from previous section.
+Clustered caches can have fixed amount of storage allocated directly from these resources.
+
+Resource pool::
+Resource pools are storage pools defined at the cache manager level that multiple caches can share.
+The real storage for these pools are carved out from server off-heap resources.
+Multiple resource pools can be allocated from the same server off-heap resource.
+Similarly, resource pools consuming storage from different server off-heap resources can be configured under the same cache manager.
+The storage available in these resource pools are strictly shared.
+That is, no cache can ask for a fixed amount of storage from these resource pools.
+If a cache needs a fixed amount of storage that it is not willing to share, then it must be allocated directly from a server off-heap resource.
+
+Here is a pictorial representation of the concepts explained above:
+
+image::storagePools.png[]
+
+Sharing of storage using resource pools here doesn't mean that the data is shared.
+That is, if two caches are using a shared resource pool, the data of each cache is still isolated.
+One can not access the data of the other.
+They are just sharing the same storage resource.
+What it means is that, when shared caches hit the capacity of their shared resource pool, and when one tries to expand, it will expand by shrinking others' storage resource.
```

Comment:
I think this sentence can be made clearer, but I also wonder if it's correct as is. If a shared pool is at capacity, then a putAll is executed on cache Foo, is it absolutely true that only the other caches pool usage will be lowered (mappings evicted) or could mappings in Foo also be removed to accommodate the new mappings.